### PR TITLE
Javascript javascript3 week3

### PR DESCRIPTION
**Pull request by Aditi-Sawardekar**
Initial Commit

### CODE REVIEW

File: javascript/javascript3/week3/screenshot.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+const displayScreenshot = document.querySelector("#image");
+
+// Save Screenshot
+const saveScreenshotButton = document.querySelector(".saveScreenshot");
+
+const getScreenshot = async (event) => {
+    // Search Input
+    const searchInput = document.querySelector("#url").value
+
+    if (!searchInput) {
```
Comment (chezzoba) [inline javascript/javascript3/week3/screenshot.js]: Could you try to ensure they have entered a valid URL using a [regular expression](https://stackoverflow.com/questions/3809401/what-is-a-good-regular-expression-to-match-a-url)?

File: javascript/javascript3/week3/screenshot.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        alert("Please enter a url");
+        return;
+    }
+
+    const HOST_URL = "https://website-screenshot6.p.rapidapi.com/screenshot"
+    const XRapidAPIKey = "f797f72385msh65df0fcdf98a7dep1ff9b3jsn9efc4dfb6fac";
```
Comment (chezzoba) [inline javascript/javascript3/week3/screenshot.js]: Oh no. What do we have here?

File: javascript/javascript3/week3/screenshot.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    const XRapidAPIHost = "website-screenshot6.p.rapidapi.com";
+    const url = searchInput;     //"https://rapidapi.com/marketplace" //"https://www.hackyourfuture.dk/" //"https://rapidapi.com/marketplace";
+    const width = 1920;
+    const height = 1080;
+
+    const options = {
+        method: 'GET',
+        headers: {
+            'X-RapidAPI-Key': 'f797f72385msh65df0fcdf98a7dep1ff9b3jsn9efc4dfb6fac',
```
Comment (chezzoba) [inline javascript/javascript3/week3/screenshot.js]: You already (sadly) have these as variables, why retype them?

File: javascript/javascript3/week3/screenshot.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    const data = await getScreenshot();
+    console.log(data);
+
+    const screenshotUrl = data.screenshotUrl;
+    console.log(screenshotUrl);
+
+    const body = { imageUrl: screenshotUrl }
+
+    try {
+        const response = await fetch(`https://crudcrud.com/api/c915ea260e2e45b3875ee6c1da488d41/blog`, {
```
Comment (chezzoba) [inline javascript/javascript3/week3/screenshot.js]: This URL should be variable(s).

**Review by chezzoba — COMMENTED** [review]
Nicely done, a few best practices are still not followed. But other than that it looks like it works.

File: javascript/javascript3/week3/screenshot.js (hunk 4 above)
Comment (Aditi-Sawardekar) [reply javascript/javascript3/week3/screenshot.js]: Made corrections.

File: javascript/javascript3/week3/screenshot.js (hunk 3 above)
Comment (Aditi-Sawardekar) [reply javascript/javascript3/week3/screenshot.js]: Made corrections in resubmission

File: javascript/javascript3/week3/screenshot.js (hunk 2 above)
Comment (Aditi-Sawardekar) [reply javascript/javascript3/week3/screenshot.js]: Recreated file, to solve this

File: javascript/javascript3/week3/saveScreenshot.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
+const BASE_URL = `https://crudcrud.com/api/${API_KEY}`
+
+// Save Screenshot
+const saveScreenshotButton = document.querySelector(".saveScreenshot");
+
+const saveScreenshot = async (event) => {
+    event.preventDefault();
+    
+    const searchInput = document.querySelector("#url").value
```
Comment (chezzoba) [inline javascript/javascript3/week3/saveScreenshot.js:9]: You still haven't validated that they have entered a valid URL (perhaps using regex)

**Review by chezzoba — APPROVED** [review]
Well done, I don't see any API keys or base URLS hardcoded anymore so that's very nice.

**Comment by Aditi-Sawardekar** [comment]
Hi Kaizad,
Thanks for reviewing my code and approving the merge request

File: javascript/javascript3/week3/getScreenshot.js [hunk 6]
```diff
@@ -18,8 +18,8 @@ const getScreenshot = async (event) => {
         alert("Please enter a url");
         return;
     } else{
-        const match = "http"
-        const regex = new RegExp(match, 'gi');        
+        const regex = /^(http(s):\/\/.)[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)$/gi;
+    
         if ((regex.test(searchInput)) === false){
```
Comment (chezzoba) [inline javascript/javascript3/week3/getScreenshot.js]: Can be `if (!regex.test(searchInput))`

**Review by chezzoba — COMMENTED** [review]
Great! The regular expression is a bit constraining but it does work. Nice job.

File: javascript/javascript3/week3/getScreenshot.js (hunk 6 above)
Comment (Aditi-Sawardekar) [reply javascript/javascript3/week3/getScreenshot.js]: Oh yes... Thanks you so much once again. I will make this change and merge the branch.

**Pull request merged**